src/main/java/org/jboss/threads/Messages.java

```diff
@@ -97,6 +97,25 @@ interface Messages extends BasicLogger {
     @LogMessage(level = Logger.Level.ERROR)
     void taskSubmitFailed(@Cause RejectedExecutionException e, Runnable task);
 
+    @Message(id = 15, value = "Thread pool has reached %d max threads in use. Performance may be impacted. Thread dump:\n"+
```

Comment:
For reference, IDs have previously been assigned to auto generated thread dump to logging functions like in eap/wildfly-core itself:

https://github.com/aogburn/wildfly-core/blob/b3cfcb741c8ced604d8895de48ec920fc130057d/controller/src/main/java/org/jboss/as/controller/logging/ControllerLogger.java#L3710

Or in activemq-artemis:

https://github.com/apache/activemq-artemis/blob/main/artemis-server/src/main/java/org/apache/activemq/artemis/core/server/ActiveMQServerLogger.java#L901

Message ids can be quite helpful to support and I and others use them in simple tooling (https://github.com/aogburn/yala) to quickly search and handle critical known message ids or sum up key information in logging.  So for a usability perspective, I'd request that it have an ID.

I think to be truly formatted exactly like a normal JDK thread dump it would probably require being generated through normal JDK utilities, like forking an external process that's calling the JDK jstack command if available (then a dump can be missed if someone has not installed the jdk-devel package or is running a JRE).

Reply:
Changed this to just one message for the thread dump messages.  I did leave one additional id to handle any exception from a dump attempt.  Is that suitable?

Reply:
@dmlloyd Just wanted to check if you'd have any further feedback here?  Thanks!

Reply:
I've been ruminating on this. For too long I guess. :) I have a couple more comments.